### PR TITLE
Adds a brain scan mode to the cloning system

### PR DESCRIPTION
**Pull request by Crazylemon64**
A tier-4 DNA scanner will now allow you to scan from the DNA of the brain of a corpse, in event of the original body getting gibbed, and will also allow a possible return route for non-standard races that doesn't involve them being "Monkey (387)" forever, anymore.
:cl:
tweak: The cloner now checks for NO_SCAN on the brain when scanning - unclonable species are now truly unclonable. You are still able to revive them with a brain transplant and a defib, however.
rscadd: A tier-4 DNA scanner linked to a cloning system will now be able to reproduce a body from a brain's DNA, in event of the original corpse being gibbed or similar.
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
I didn't think you were going to go for *this* angle of things; being able to negate a species intentional weakness at 20 minutes in seems..problematic (particularly for Vox). I knew this was coming, but I didn't think it was going to be from the angle of "violate species flags", and was more of the "bebuild them so they have a proper name".

Doesn't this also hypothetically mean uh...you could clone IPCs?

**Comment by Crazylemon64** [comment]
It still does all the old checks! You'll need to get them in a new body for it to recognize them! Should I add another check, for the species, so that if the species type can't normally be cloned, that they take the species of the host mob? I mean, as it stands, those species CAN be cloned, but you just need to do a brain transplant first - the result is that they are no longer that species. Is it desired that once cloned, they lose all chance to be that species again?

**Comment by VampyrBytes** [comment]
Question - What happens when a traitor steals a brain of someone who's had a brain scan? I've only had a quick glance, but I think they would fail if the scan had been used to bring them back.

Personally I think death is easy enough to get around as it is without making it easier, not to mention I can already see the queue's that would form at the scanner once it's upgraded

Oh, and if you want to fix the name issue, just give the brain a real name var that's set up from it's mob in it's new() and use that to set the name in the cloner

**Comment by Aurorablade** [comment]
@Crazylemon64 "So no, this won't let you clone IPCs, nor will it let you directly bypass the cloning restrictions on other species. It WILL, however, produce a body of the species of the brain - I take it this is undesired, yes?" only for Vox and Slime people. typically they are uncloneable. I think plasmamen are too...

**Comment by NTSAM** [comment]
Plasmamen can be cloned, they just usually combust upon completion. Still possible to do if you're quick.

**Comment by Crazylemon64** [comment]
> Question - What happens when a traitor steals a brain of someone who's had a brain scan?

The same as if the traitor steals the brain of someone who's had a regular cloning scan - since the target client isn't a brainmob in the traitor's inventory, they fail the mission either way - provided the data is actually cloned from, of course.

> Personally I think death is easy enough to get around as it is without making it easier

This PR doesn't change how easy players are to bring back from the dead - the brain must still be in a corpse viable for standard scanning methods. This just changes the resulting body - gives it the DNA of the original person, rather than that of the body the brain was placed in.

> I can already see the queue's that would form at the scanner once it's upgraded

This feature does literally nothing for people who are still in their original bodies.


Anyways, it seems like producing bodies of non-clonable species is getting negative reception; I'll include a check to instead make it the species of the host body, if the brain's species is normally unclonable.

**Comment by Crazylemon64** [comment]
Ah, if you have objections to non-clonable species being cloned, should I add a check to the cloners to stop cloning attempts that use the brain of a non-clonable species, even if the body is otherwise compatible, and a brain scan isn't being done?

**Comment by VampyrBytes** [comment]
Thanks for the answers @Crazylemon64 As I said, I'd only glanced at the code, and was still half asleep when I posted, it was just possible issues that sprung to mind.

And yes, I would like to see a check on the brain's species type in the cloner. The no_scan is put in their to help balance the races, and shouldn't really have a work around

**Comment by fludd12** [comment]
Er, if that happened, wouldn't it be impossible to revive unclonable races, even with a brain transplant? Losing the race you were seems like enough of a downside, to me.

**Comment by Crazylemon64** [comment]
@fludd12 You can still transplant the brain into a fresh body, and defib, to bring them back

**Comment by Regen1** [comment]
@Crazylemon64 Yes, but that puts a lot of effort behind it, as it should.

**Comment by Crazylemon64** [comment]
I'd like this not to be merged until <a href=http://nanotrasen.se/forum/viewtopic.php?f=12&p=54599>this thread</a> reaches a definite consensus, or fizzles to the point where I determine that no one cares - I'll post again when it's reached that point.

**Comment by Crazylemon64** [comment]
Ah, if merged, this will be yet another thing for #2903 to deal with, so this should probably wait until that one's done.

**Comment by Aurorablade** [comment]
YOU PEOPLE STOP WAITING UNTIL I AM DONE

**Comment by Crazylemon64** [comment]
Ok!

**Comment by Crazylemon64** [comment]
The thread seems to be OK on this, feel free to take away the DNM flag and merge away

File: html/changelogs/crazylemon-cloning_brain_scan.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+author: Crazylemon
+delete-after: True
+changes: 
+  - rscadd: "A cloner with a tier-4 scanner will now be able to scan the brains of bodies within it, giving a chance for people who've otherwise lost their original corpse and identity."
```
Comment (Fox-McCloud) [inline html/changelogs/crazylemon-cloning_brain_scan.yml]: no file chanelogs, [remember](https://github.com/ParadiseSS13/Paradise/pull/3420#issuecomment-175492813)

Comment (Crazylemon64) [inline html/changelogs/crazylemon-cloning_brain_scan.yml]: I know, but this one was in place before you gave me the notice - do you want this updated?

**Comment by Crazylemon64** [comment]
Made it more of a mode rather than a button so that it works cool with autoprocess

**Pull request merged**